sympy/core/tests/test_relational.py

```diff
@@ -1187,3 +1188,94 @@ def _eval_is_ge(lhs, rhs):
     assert is_gt(PowTest(3, 9), PowTest(3,2))
     assert is_le(PowTest(3, 2), PowTest(3,9))
     assert is_lt(PowTest(3, 2), PowTest(3,9))
+
+def test_issue_20584():
+    # TODO: cases with multiple valid results should be removed once the
+    #  relations evaluate to the most simplified values (first in valid,
+    #  if present)
+    xs = {None: x,
+          'finite': Symbol('x', finite=True),
+          'infinite': Symbol('x', infinite=True)}
+
+    cases = []
+    expr_uneval = Lt(Abs(x), oo, evaluate=False)
+    cases.append(
+        (expr_uneval, {
+            None: [expr_uneval],
+            'finite': [S.true, expr_uneval],
+            'infinite': [S.false, expr_uneval],
+        })
+    )
+    expr_uneval = Lt(oo, Abs(x), evaluate=False)
+    cases.append(
+        (expr_uneval, {
+            None: [S.false, expr_uneval],
+            'finite': [S.false, expr_uneval],
+            'infinite': [S.false, expr_uneval],
+        })
+    )
+    expr_uneval = Le(Abs(x), oo, evaluate=False)
+    cases.append(
+        (expr_uneval, {
+            None: [S.true, expr_uneval],
+            'finite': [S.true, expr_uneval],
+            'infinite': [S.true, expr_uneval],
+        })
+    )
+    expr_uneval = Le(oo, Abs(x), evaluate=False)
+    cases.append(
+        (expr_uneval, {
+            None: [expr_uneval],
+            'finite': [S.false, expr_uneval],
+            'infinite': [S.true, expr_uneval],
+        })
+    )
+    expr_uneval = Gt(Abs(x), oo, evaluate=False)
+    cases.append(
+        (expr_uneval, {
+            None: [S.false, expr_uneval],
+            'finite': [S.false, expr_uneval],
+            'infinite': [S.false, expr_uneval],
+        })
+    )
+    expr_uneval = Gt(oo, Abs(x), evaluate=False)
+    cases.append(
+        (expr_uneval, {
+            None: [expr_uneval],
+            'finite': [S.true, expr_uneval],
+            'infinite': [S.false, expr_uneval],
+        })
+    )
+    expr_uneval = Ge(Abs(x), oo, evaluate=False)
+    cases.append(
+        (expr_uneval, {
+            None: [expr_uneval],
+            'finite': [S.false, expr_uneval],
+            'infinite': [S.true, expr_uneval],
+        })
+    )
```

Comment:
I find this format for the tests quite complicated. If we were to write this out long form would it be something like:
```python
# Define these once at the top:
xi = Symbol('xi', infinite=True)
xf = Symbol('xf', finite=True)

assert unchanged(Ge, Abs(x), oo)
assert unchanged(Ge, Abs(xf), oo)  # Ideally S.false
assert Ge(xi, oo) is S.false
```